### PR TITLE
Allow to_json for Log::Context

### PR DESCRIPTION
**Pull request by paulcsmith**
Pretty self explanatory :D As discussed in https://forum.crystal-lang.org/t/log-module-suggestions/1962/5

### CODE REVIEW

**Comment by paulcsmith** [comment]
Closing in favor of https://github.com/crystal-lang/crystal/pull/9092/files which I did not see at first :)

**Comment by bcardiff** [comment]
I think this PR does the json integration as @asterite and I stated we preferred. 

I would rather reopen and merge this as is.

**Comment by paulcsmith** [comment]
Sounds good to me!

File: src/log/json.cr [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require "json"
+
+class Log::Context
+  # Returns Log::Context as JSON value.
```
Comment (Sija) [inline src/log/json.cr:4]:
```suggestion
  # Returns `Log::Context` as JSON value.
```

File: src/log/context.cr [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Immutable structured context information for logging.
 #
 # See `Log.context`, `Log.context=`, `Log::Context#clear`, `Log::Context#set`, `Log.with_context`.
+#
+# > If you'd like to format the context as JSON remember to `require "log/json"`.
```
Comment (Sija) [inline src/log/context.cr:5]:
```suggestion
# NOTE: If you'd like to format the context as JSON, remember to `require "log/json"`.
```

File: src/log/json.cr [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  #
+  # NOTE: `require "log/json"` is required to opt-in to this feature.
+  #
+  # ```
+  # log_entry.context.to_json # => {"user_id" => 1}
```
Comment (Sija) [inline src/log/json.cr:9]:
```suggestion
  # log_entry.context.to_json # => "{\"user_id\":1}"
```

**Comment by bcardiff** [comment]
We would need one more approval here.

**Comment by straight-shoota** [comment]
Thanks @paulcsmith

**Pull request merged**